jib-core/src/main/java/com/google/cloud/tools/jib/registry/ManifestPusher.java

```diff
@@ -58,6 +60,24 @@ public Void handleResponse(Response response) {
     return null;
   }
 
+  @Override
+  public Void handleHttpResponseException(HttpResponseException httpResponseException)
+      throws HttpResponseException, RegistryErrorException {
+    if (httpResponseException.getStatusCode() == HttpStatusCodes.STATUS_CODE_NOT_FOUND) {
+      /*
+       * This error could happen if the repository doesn't exist (which should fail at BLOB-push).
+       * The more likely reason is that the image is pushing to a tag with backslashes in it, in
+       * which part of the tag becomes part of the repository.
+       */
+      throw new RegistryErrorExceptionBuilder(getActionDescription(), httpResponseException)
+          .addReason(
+              "repository name not known to registry (perhaps you are using an invalid tag - tags cannot contain backslashes)")
```

Comment:
Is the repository name available here? If so it could be in the error message and you could tell for sure if it contains slashes.

Reply:
I've tried this on GCR and it produces a correct error message already (about using an invalid tag - this error message will show up as part of the wrapped HttpResponseException), but on ECR, it simply gives a 404 and not in the expected Docker registry 404 format.

Reply:
If we know their tag here already on our own, from `imageTag`, can we just tell them that their tag contains backslashes if it does?

Reply:
and if it doesn't, then we print the registry from `registryEndpointProperties.getImageName()` ?

Reply:
If their `imageTag` contains backslashes, we might want to just tell them that before entering the build process. Maybe we should just add a warning in the Maven plugin if the tag configuration contains backslashes?

Reply:
If tags with `\` are illegal, perhaps manifest pusher might want to throw an illegal arugment exception in the constructor?

Reply:
I think it might be good to keep `ManifestPusher` decoupled from any logic about the formatting of the image tag, since that is specific to the implementation of different Docker registries. Also, an `IllegalArgumentException` would mean an additional logic in `BuildImageMojo` to catch that exception and give the user a more meaningful error message. I added some validation in the `BuildImageMojo` that validates its configuration - currently it just logs some errors but doesn't fail, but perhaps it should fail? I can also remove this custom error message in `ManifestPusher` since the one in the Mojo is more helpful.